src/monitor/node_active_protocol.c

```diff
@@ -2015,38 +2071,43 @@ set_node_replication_quorum(PG_FUNCTION_ARGS)
 			GetPrimaryNodeInGroup(currentNode->formationId,
 								  currentNode->groupId);
 
-		if (primaryNode == NULL)
+		/*
+		 * If we allow setting changes during APPLY_SETTINGS we open the door
+		 * for race conditions where we can't be sure that the latest changes
+		 * have been applied.
+		 *
+		 * If we don't currently have a primary node anyway, we can just
+		 * proceed with the change.
+		 */
+		if (primaryNode &&
```

Comment:
Maybe move this check to a function, since it's used when setting both replication_quorum and candidate_priority

Reply:
I'm not sure, I feel like it's fair game to have a simple two step conditions copy/pasted in here.